src/main/java/com/google/devtools/build/lib/query2/PostAnalysisQueryEnvironment.java

```diff
@@ -631,4 +634,28 @@ public Iterable<BuildConfiguration> getConfigurations() {
       }
     }
   }
+
+  public abstract boolean shouldOrderResults();
+
+  public List<T> orderResults(Iterable<T> resultList)
+      throws InterruptedException {
+
+    Digraph<T> graph = new Digraph<>();
+    ImmutableSet<T> resultSet = ImmutableSet.copyOf(resultList);
```

Comment:
Could we save us this copy by having two passes on `resultList`? One creating all the nodes and one suing `getNode`/`getNodeMaybe` to add the dependencies.

Reply:
I think so, will give that way a shot.